### PR TITLE
update changelog for v0.3.0

### PR DESCRIPTION
**Pull request by endophage**
Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
++ Update project to Go 1.6.1
++ Reorganize vendoring to meet Go 1.6+ standard. Still using Godeps to manage vendored packages
++ Add targets by hash, no longer necessary to have the original target data available
++ Active Key ID verification during signature verification
++ Switch all testing from assert to require, reduces noise in test runs
++ Use alpine based images for smaller downloads and faster setup times
++ Clean up out of data signatures when re-signing content
++ Set cache control headers on HTTP responses from Notary Server
++ Add sha512 support for targets
++ Move SQLite dependency to tests only so it doesn't get build into official images
```
Comment (riyazdf) [inline CHANGELOG.md]: nit typo `built`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
 # Changelog
 
+## [v0.3.0](https://github.com/docker/notary/releases/tag/v0.3.0) 5/11/2016
++ Root rotations
++ Delegation support
```
Comment (riyazdf) [inline CHANGELOG.md]: I'm a little fuzzy on what this entails, since delegation support was 0.2?

**Comment by riyazdf** [comment]
Couple of PRs that I think aren't in the changelog (though I might've missed it!), and that we might want to be included:

`Add environment variable for automating delegation key passphrase entry`
https://github.com/docker/notary/pull/645

`Update formatting for delegation list output to be multiline for keys and paths`
https://github.com/docker/notary/pull/641

`Update delegation behavior to not fallback to parent roles if specified role does not exist`
https://github.com/docker/notary/pull/616

`Further restrict JWT token permissions to remove unnecessary actions`
https://github.com/docker/notary/pull/594

**Comment by cyli** [comment]
1.  Should we be including the PRs where possible in this changelog like docker does?  I'd be happy to make the change.

2.  Should we split these up into the major updates, minor updates, and bugfixes like in https://github.com/docker/notary/issues/738#issuecomment-218598317?

**Comment by endophage** [comment]
@cyli I stuck with the existing format we already had for other changelog items. Happy to go either way.

**Comment by endophage** [comment]
@riyazdf added notes for 3/4 of those PRs you listed. Don't particularly want to call out #616. Feel like it's likely to be more confusing than useful to include it.

**Comment by riyazdf** [comment]
@endophage makes sense for 616 -- I'll copy this over to the draft release, LGTM!

**Comment by riyazdf** [comment]
actually, I think we should also update: https://github.com/docker/notary/blob/master/docs/changelog.md

I can do this in a separate PR if you'd like, I think this PR is good to go as is

**Comment by cyli** [comment]
I'm fine with the format remaining as is, since we don't have a good other changelog process right now.  LGTM!

**Pull request merged**